subprojects/core/src/main/java/org/gradle/api/tasks/bundling/Zip.java

```diff
@@ -109,39 +93,16 @@ public void setZip64(boolean allowZip64) {
      * This means you should not enable this property if you are building JARs to be used with Java 6 and earlier runtimes.
      */
     @Input
-    @ToBeReplacedByLazyProperty
-    public boolean isZip64() {
-        return allowZip64;
-    }
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getZip64();
```

Comment:
There is a smoke test failure:
>Execution failed for task ':shadowJar'.
  > Could not find matching constructor for: com.github.jengelman.gradle.plugins.shadow.internal.DefaultZipCompressor(org.gradle.api.internal.provider.DefaultProperty, Integer)


https://github.com/GradleUp/shadow/blob/b35e6904a1dfa541fc147d46e9ba456fdbe3f176/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/GradleVersionUtil.groovy#L18
```groovy
                return new DefaultZipCompressor(jar.zip64, ZipOutputStream.DEFLATED)
```